lib/ansible/modules/packaging/os/snap.py

```diff
@@ -0,0 +1,165 @@
+        choices: [ absent, present ]
+    classic:
+        description:
+            - Confinement policy. The classic confinment allows a snap to have
+              the same level of access to the system as "classic" packages,
+              like those managed by APT. This option corresponds to the --classic argument.
+        type: bool
+        required: false
+        default: False
+
+author:
+    - Victor Carceler (vcarceler@iespuigcastellar.xeill.net)
+    - Stanislas Lange (angristan) <angristan@pm.me>
+'''
+
+EXAMPLES = '''
+# Install "foo" snap
+- name: Install foo
+  snap:
+    name: foo
+
+# Remove "foo" snap
+- name: Remove VLC
+  snap:
+    name: foo
+    state: absent
+
+# Install a snap with classic confinement
+- name: Install "foo" with option --classic
+  snap:
+    name: foo
+    classic: True
+'''
+
+RETURN = '''
+msg:
+    description: An informative message on the task's output
+    type: str
+    returned: always
+    sample: "The snap 'foo' has been installed"
+'''
+
+from ansible.module_utils.basic import AnsibleModule
+
+
+def snap_exists(module, snap_name):
+    snap_path = module.get_bin_path("snap", True)
+    cmd = "%s info %s" % (snap_path, snap_name)
+    rc, out, err = module.run_command(cmd, check_rc=False)
+
+    return rc, out, err
+
+
+def is_snap_installed(module, snap_name):
+    snap_path = module.get_bin_path("snap", True)
+    cmd = "%s list %s" % (snap_path, snap_name)
+    rc, out, err = module.run_command(cmd, check_rc=False)
+
+    return rc, out, err
+
+
+def install_snap(module, snap_name):
+    rc, out, err = is_snap_installed(module, snap_name)
+    if rc == 0:
+        module.exit_json(msg="The snap '%s' is already installed" % str(snap_name))
+
+    if module.params['classic']:
+        classic = '--classic'
+    else:
+        classic = ''
+
+    snap_path = module.get_bin_path("snap", True)
+    cmd = "%s install %s %s" % (snap_path, snap_name, classic)
+
+    # Actually install the snap
+    rc, out, err = module.run_command(cmd, check_rc=False)
+
+    if rc == 0:
+        # snap has been installed
+        module.exit_json(msg="The snap '%s' has been installed" % str(snap_name), changed=True)
+    else:
+        # Something went wrong
+        module.fail_json(msg=err)
+
+
+def remove_snap(module, snap_name):
+    rc, out, err = is_snap_installed(module, snap_name)
+    if rc != 0:
+        module.exit_json(msg="The snap '%s' is not installed" % str(snap_name))
+
+    snap_path = module.get_bin_path("snap", True)
+    cmd = "%s remove %s" % (snap_path, snap_name)
+
+    # Actually remove the snap
+    rc, out, err = module.run_command(cmd, check_rc=False)
+
+    if rc == 0:
+        # snap has been removed
+        module.exit_json(msg="The snap '%s' has been removed" % str(snap_name), changed=True)
+    else:
+        # Something went wrong
+        module.fail_json(msg=err)
+
+
+def main():
+    module_args = dict(
+        name=dict(type='str', required=True),
```

Comment:
I recommend you to make a list out of it `type='list'`, it'll allow you to do
```yaml
- snap:
    name:
    - package1
    - package2
    ...
```
instead of looping over task execution, which is typically slower

Reply:
OK! Then, if I loop over my list, how should I handle exits?

Reply:
It depends. If `snap` command supports installing several modules at once you can just pass them all (concatenate) when calling it.

Reply:
Here's what I did: https://github.com/ansible/ansible/pull/44939/commits/42acd3d57d2ce962bc93389ea16b351f66fc9990